synapse/cores/sqlite.py

```diff
@@ -502,13 +538,90 @@ def _initCorQueries(self):
     def _checkForTable(self, name):
         return len(self.select(self._q_istable, name=name))
 
+    def _initCorTables(self, table):
+
+        revs = [
+            (0, self._rev0)
+        ]
+
+        max_rev = max([rev for rev, func in revs])
+        vsn_str = 'syn:core:{}:version'.format(self.getCoreType())
+
+        if not self._checkForTable(table):
+            # We are a new cortex, stamp in tables and set
+            # blob values and move along.
+            self._initCorTable(table)
+            self.setBlobValu(vsn_str, max_rev)
+            return
+
+        # Strap in the blobstore if it doesn't exist - this allows us to have
+        # a helper which doesn't have to care about queries agianst a table
+        # which may not exist.
+        blob_table = table + '_blob'
+        if not self._checkForTable(blob_table):
+            with self.getCoreXact() as xact:
+                xact.cursor.execute(self._q_init_blobtable)
+                xact.cursor.execute(self._q_init_blobtable_idx)
+
+        # Apply storage layer revisions
+        self._revCorVers(revs)
+
+    def _revCorVers(self, revs):
```

Comment:
this method ( and the implementation above ) feel like they should be unified rather than implemented in each individual storage layer ( especially since they're basically the same... )